Yank/analyze.py

```diff
@@ -32,27 +35,1145 @@
+        parameters.
+
+        This function is hidden from the user unless they really, really need to create their own mbar object
+
+        energy_matrix : array of numpy.float64, optional, default=None
+           Reduced potential energies of the replicas; if None, will be extracted from the ncfile
+        samples_per_state : array of ints, optional, default=None
+           Number of samples drawn from each kth state; if None, will be extracted from the ncfile
+
+        """
+
+        # Delete observables cache since we are now resetting the estimator
+        for observable in self.observables:
+            self._computed_observables[observable] = None
+
+        # Initialize MBAR (computing free energy estimates, which may take a while)
+        logger.info("Computing free energy differences...")
+        mbar = MBAR(energy_matrix, samples_per_state)
+
+        self._mbar = mbar
+
+    def _combine_phases(self, other, operator='+'):
+        phases = [self]
+        names = []
+        signs = [self._sign]
+        # Reset self._sign
+        self._sign = '+'
+        if self.name is None:
+            names.append(generate_phase_name(self, []))
+        else:
+            names.append(self.name)
+        if isinstance(other, MultiPhaseAnalyzer):
+            new_phases = other.phases
+            new_signs = other.signs
+            new_names = other.names
+            final_new_names = []
+            for name in new_names:
+                other_names = [n for n in new_names if n != name]
+                final_new_names.append(generate_phase_name(name, other_names + names))
+            names.extend(final_new_names)
+            for new_sign in new_signs:
+                if operator != '+' and new_sign == '+':
+                    signs.append('-')
+                else:
+                    signs.append('+')
+            phases.extend(new_phases)
+        elif isinstance(other, YankPhaseAnalyzer):
+            names.append(generate_phase_name(other.name, names))
+            if operator != '+' and other._sign == '+':
+                signs.append('-')
+            else:
+                signs.append('+')
+            # Reset the other's sign if it got set to negative
+            other._sign = '+'
+            phases.append(other)
+        else:
+            baseerr = "cannot {} 'YankPhaseAnalyzer' and '{}' objects"
+            if operator == '+':
+                err = baseerr.format('add', type(other))
+            else:
+                err = baseerr.format('subtract', type(other))
+            raise TypeError(err)
+        phase_pass = {'phases': phases, 'signs': signs, 'names': names}
+        return MultiPhaseAnalyzer(phase_pass)
+
+    def __add__(self, other):
+        return self._combine_phases(other, operator='+')
+
+    def __sub__(self, other):
+        return self._combine_phases(other, operator='-')
+
+    def __neg__(self):
```

Comment:
Kind of, I could not find a way to pass the negative operator in a non-preserving manner without doing a deepcopy which would be super memory inefficient and break the ability for adjusting `SinglePhase` `reference_state`'s that propagate to all `MultiPhase` analyzers which use them, which is why all the functions which combine phases actually reverse them.

E.g.
```python
>>> x = RepexAnalyzer(...)
>>> print(x.sign)
  '+'
>>> -x
>>> print(x.sign)
  '-'
```
which is not the exact behavior we want. But I could not find a way to detect that both negative operator has been used in a more complex operation, and not as strait re-assignment. (can't overload the `=` operator)